starboard/android/apk/app/src/main/java/dev/cobalt/media/VideoFrameReleaseTimeHelper.java

```diff
@@ -199,10 +200,18 @@ protected void onSynced() {
   private boolean isDriftTooLarge(long frameTimeNs, long releaseTimeNs, double playbackRate) {
     long elapsedFrameTimeNs = frameTimeNs - syncFramePresentationTimeNs;
     long elapsedReleaseTimeNs = releaseTimeNs - syncUnadjustedReleaseTimeNs;
-    return Math.abs(elapsedReleaseTimeNs - elapsedFrameTimeNs / playbackRate)
+    return Math.abs(
+            elapsedReleaseTimeNs - adjustFrameTimeForPlaybackRate(elapsedFrameTimeNs, playbackRate))
         > MAX_ALLOWED_DRIFT_NS;
   }
 
+  private long adjustFrameTimeForPlaybackRate(long frameTimeNs, double playbackRate) {
+    // We support playback rate [0, 2]. So, a multiplier of 10 is good here to keep precision.
```

Comment:
Can you elaborate (and maybe provide example numbers) on why this is required to improve the precision of calculation?

Reply:
Double has 52 bits for number precision while long has 63 bits. The timestamp is in ns, sometimes it might be very large. If we convert the numbers to double and do the calculation, it might lose a few precision.

Reply:
Let's add your description to the comment.  Also, since long has 63 bits of precision, does it have enough precision for us to use `frameTimeNs * PLAYBACK_RATE_ADJUSTMENT_MULTIPLIER / multipliedPlaybackRate;`?